library/sha256.c

```diff
@@ -389,10 +397,16 @@ static const unsigned char sha256_test_sum[6][32] =
 int mbedtls_sha256_self_test( int verbose )
 {
     int i, j, k, buflen, ret = 0;
-    unsigned char buf[1024];
+    unsigned char *buf;
     unsigned char sha256sum[32];
     mbedtls_sha256_context ctx;
 
+    buf = mbedtls_calloc(1024, sizeof(unsigned char));
+    if (0 == buf)
+    {
+        return 1;
```

Comment:
I think it would also be idea if we use the already defined error codes, such as `MBEDTLS_ERR_CIPHER_ALLOC_FAILED` instead of hard-coding the value.